src/ape/utils/misc.py

```diff
@@ -543,6 +544,37 @@ def wrapped(*args, **kwargs):
     return wrapper
 
 
+_MOD_T = TypeVar("_MOD_T")
+
+
+def as_our_module(cls_or_def: _MOD_T, doc_str: Optional[str] = None) -> _MOD_T:
+    """
+    Ape sometimes reclaims definitions from other packages, such as
+    class:`~ape.types.signatures.SignableMessage`). When doing so, the doc str
+    may be different than ours, and the module may still refer to
+    the original package. This method steals the given class as-if
+    it were ours. Logic borrowed from starknet-py.
```

Comment:
Might be good to link to that code

Reply:
[12bd6f2](https://github.com/ApeWorX/ape/pull/2175/commits/12bd6f2f5a2d53678fa59f6d72b44da1cf78d938)